pkg/apis/clusterregistry/v1alpha1/types.go

```diff
@@ -23,16 +23,89 @@ import (
 // +genclient
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
 // Information about a cluster in a cluster registry.
 type Cluster struct {
 	metav1.TypeMeta `json:",inline"`
 	// Standard object's metadata.
 	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
 	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
I don't think this should be optional

Reply:
This makes conceptual sense to me, but it looks like many (most?) Kubernetes API objects do mark the ObjectMeta as optional.

https://github.com/kubernetes/api/blob/master/core/v1/types.go
https://github.com/kubernetes/api/blob/master/apps/v1beta2/types.go

I'm not sure I want to go against that convention.

Reply:
bizarre and unexpected, but fine for now